hw_diag/views/diagnostics.py

```diff
@@ -111,3 +114,46 @@ def version_information():
     }
 
     return response
+
+
+@DIAGNOSTICS.route('/v1/add-gateway-txn', methods=['POST'])
+def add_gateway_txn():
+    """
+    Generates an add_gateway_txn if a destination name and wallets are defined and valid.
+    Diagnostics report will be in the format below if successful.
+
+    https://docs.helium.com/mine-hnt/full-hotspots/become-a-maker/hotspot-integration-testing/#generate-an-add-hotspot-transaction
+    {
+        errors: [],
+        DESTINATION_ADD_GATEWAY_TXN_KEY: {
+            "address": "11TL62V8NYvSTXmV5CZCjaucskvNR1Fdar1Pg4Hzmzk5tk2JBac",
+            "fee": 65000,
+            "owner": "14GWyFj9FjLHzoN3aX7Tq7PL6fEg4dfWPY8CrK8b9S5ZrcKDz6S",
+            "payer": "138LbePH4r7hWPuTnK6HXVJ8ATM2QU71iVHzLTup1UbnPDvbxmr",
+            "staking fee": 4000000,
+            "txn": "CrkBCiEBrlImpYLbJ0z0hw5b4g9isRyPrgbXs9X+RrJ4pJJc9MkS..."
+        }
+    }
+    """
+
+    shipping_destination_with_signature = request.get_data()
```

Comment:
Have you verified that this works with your test cases where data is being sent as multipart form data (normal HTTP Form POST)? AFAIK `get_data()` works when the POST data does not contain a mime-type/content-type. If we're passing `multipart/form-data' and content type this might cause get_data not to work as expected. At the very minimum based on the docs [here](https://tedboy.github.io/flask/generated/generated/flask.Request.get_data.html) we need to set `parse_form_data` to `True` like `request.get_data(parse_form_data=True)`.

I feel like we should double check and test it. Probably will do that later today when we do some manual integration tests.

Reply:
Got it. Let me update a bit and I will ensure a request without content type should work and also **the test case** should be without a content type.